bees_algorithm.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+from numpy import genfromtxt
+
+from dataclasses import dataclass, field
+from typing import List, Tuple
+
+from nutrinator.generator import Generator
+from nutrinator.generator import GeneratorConfig as GenConf
+from nutrinator.nutrionator import Nutrinator
+
+@dataclass
+class BeesConfig:
+    n: int = 10
+    m: int = 5
+    e: int = 2
+    nsp: int = 15
+    nep: int = 6
+
+
+def stop_criterion(i):
+    return i < 10
+
+
+def generate_new_results(results: List[Tuple], result_evals: List[float],
+                         generator: Generator,
+                         conf: BeesConfig = BeesConfig()):
+    best_results_ind = np.argsort(result_evals)
+
+    m_best_results = [results[i] for i in best_results_ind[:conf.m]]
+    e_best_results = m_best_results[:conf.e]
+    nsp_neighbouring_results = generator.generate_neighbours(m_best_results, conf.nsp)
+    nep_neighbouring_results = generator.generate_neighbours(e_best_results, conf.nep)
+    random_results = [generator.generate_days() for _ in range(conf.n - conf.m)]
+
+    return m_best_results + \
+           nsp_neighbouring_results + nep_neighbouring_results + \
+           random_results
+
+
+# n - number of initial results
+# m - number of good bees
+# e - number of elite bees
+# d - number of days
+def bees_algorithm(generator: Generator, fitted_nutrinator: Nutrinator, b_conf: BeesConfig = BeesConfig()):
+    results = [generator.generate_days() for _ in range(b_conf.n)]
+    result_evaluations = [fitted_nutrinator.compute(A, P) for A, P in results]
+    i = 0
+    while stop_criterion(i):
+        results = generate_new_results(results, result_evaluations, generator, b_conf)
+        result_evaluations = [fitted_nutrinator.compute(A, P) for A, P in results]
+        print(len(results))
```

Comment:
Total number of solutions should be `n` not `n + nep + nsp`

- n - total number of bees
- `nep * e` - number of bees from elite solutions
- `nsp * m - nep * e` - number of bees from selected solutions (we select elite bees from within selected bees)
- `n - (nsp * m)` - number of random bees

Reply:
OK, makes sense. I'm still concerned about the pseudocode in the analysis. Please consider making it a little more clear.
![obraz](https://user-images.githubusercontent.com/58188916/170546301-36da0ec6-349b-460d-9197-fd3dcbfccc20.png)